### PR TITLE
added order functionality

### PR DESCRIPTION
**Pull request by suprun-anna**
created order and orderItem models, dtos, mappers, repositories, service, controller, liquibase files for creating tables

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        order = orderFromDb;
+        order.setOrderItems(orderItems);
+
+        order.setTotal(order.getOrderItems().stream()
+                .map(OrderItem::getPrice)
+                .reduce(BigDecimal.ZERO, BigDecimal::add));
+
+        shoppingCartService.clearShoppingCart(cartFromDb.getId());
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
```
Comment (MentorMA) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java]: Please divide some logic to separate method, cause this method too big

**Review by MentorMA — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -26,4 +26,12 @@ databaseChangeLog:
...
\ No newline at end of file
+      file: /db/changelog/changes/14-insert-books-categories.yaml
+  - include:
+      file: /db/changelog/changes/15-create-orders-table.yaml
+  - include:
+      file: /db/changelog/changes/16-create-order-items-table.yaml
+  - include:
+      file: /db/changelog/changes/17-insert-shopping-cart.yaml
+  - include:
+      file: /db/changelog/changes/18-insert-cart-items.yaml
```
Comment (okuzan) [inline src/main/resources/db/changelog/db.changelog-master.yaml]:
```suggestion
      file: /db/changelog/changes/18-insert-cart-items.yaml
      
```

File: src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -51,4 +51,11 @@ public UserResponseDto getByEmail(String email) {
         return user.map(userMapper::toDto).orElseThrow(
                 () -> new RuntimeException("Can't find user by email " + email));
     }
+
+    @Override
+    public User getById(Long id) {
+        Optional<User> user = userRepository.findById(id);
+        return user.orElseThrow(
+                () -> new RuntimeException("Can't find user by id= " + id));
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java]: combine into one line, throw EntityNotFound

File: src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -51,4 +51,11 @@ public UserResponseDto getByEmail(String email) {
         return user.map(userMapper::toDto).orElseThrow(
                 () -> new RuntimeException("Can't find user by email " + email));
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java]: throw EntityNotFoundException

File: src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    @Override
+    public OrderDto update(UpdateOrderStatusRequestDto requestDto, Long orderId) {
+        Optional<Order> optionalOrder = orderRepository.findById(orderId);
+        if (optionalOrder.isPresent()) {
+            Order order = optionalOrder.get();
+            order.setStatus(requestDto.status());
+            return orderMapper.toDto(order);
+        } else {
+            throw new EntityNotFoundException("Can't find order by orderId=" + orderId);
+        }
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java]: [throw early](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement), unnest rest of your code

File: src/main/java/mate/academy/bookstore/mapper/OrderMapper.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import mate.academy.bookstore.config.MapperConfig;
+import mate.academy.bookstore.dto.order.OrderDto;
+import mate.academy.bookstore.model.Order;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Mappings;
+
+@Mapper(config = MapperConfig.class, uses = OrderItemMapper.class)
+public interface OrderMapper {
+    @Mappings({
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/mapper/OrderMapper.java]:
```suggestion
```
you can just stack them one by one

File: src/main/java/mate/academy/bookstore/mapper/OrderItemMapper.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.Mappings;
+import org.mapstruct.Named;
+
+@Mapper(config = MapperConfig.class)
+public interface OrderItemMapper {
+    @Mappings({
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/mapper/OrderItemMapper.java]: same

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by MentorMA — APPROVED** [review]

**Pull request merged**